lib/ClangImporter/ImportMacro.cpp

```diff
@@ -319,9 +394,68 @@ static ValueDecl *importMacro(ClangImporter::Implementation &impl,
 
       llvm::APSInt value{ base->getValue() << shift->getValue(),
                           clangTy->isUnsignedIntegerType() };
-      return impl.createConstant(name, DC, type, clang::APValue(value),
+      return createMacroConstant(impl, macro, name, DC, type,
+                                 clang::APValue(value),
                                  ConstantConvertKind::Coerce, /*static=*/false,
                                  ClangN);
+    // Check for a expression of the form (FLAG1 | FLAG2), (FLAG1 & FLAG2),
+    // (FLAG1 || FLAG2), or (FLAG1 || FLAG2)
+    } else if (tokenI[0].is(clang::tok::identifier) &&
+               isBinaryOperator(tokenI[1]) &&
+               tokenI[2].is(clang::tok::identifier)) {
+      auto firstID = tokenI[0].getIdentifierInfo();
+      auto secondID = tokenI[2].getIdentifierInfo();
+
+      if (firstID->hasMacroDefinition() && secondID->hasMacroDefinition()) {
+        auto firstMacroInfo = impl.getClangPreprocessor().getMacroInfo(firstID);
+        auto secondMacroInfo = impl.getClangPreprocessor().getMacroInfo(
+                                                                      secondID);
+        auto firstIterator = impl.ImportedMacroConstants.find(firstMacroInfo);
+        if (firstIterator == impl.ImportedMacroConstants.end()) {
+          return nullptr;
+        }
+        auto secondIterator = impl.ImportedMacroConstants.find(secondMacroInfo);
+        if (secondIterator == impl.ImportedMacroConstants.end()) {
+          return nullptr;
+        }
+
+        auto firstConstant = firstIterator->second;
+        auto secondConstant = secondIterator->second;
+        auto firstValue = firstConstant.first;
+        auto secondValue = secondConstant.first;
+        if (!firstValue.isInt() || !secondValue.isInt()) {
+          return nullptr;
+        }
+
+        auto firstInteger = firstValue.getInt();
+        auto secondInteger = secondValue.getInt();
+        auto type = firstConstant.second;
+
+        clang::APValue value;
+        if (tokenI[1].is(clang::tok::pipe)) {
+          value = clang::APValue(firstInteger | secondInteger);
```

Comment:
This (and the `&` case below) will fail (trap) if the integers have different bit-widths.